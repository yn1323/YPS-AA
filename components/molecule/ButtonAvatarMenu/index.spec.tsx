import { mountWithTheme } from '@test/TestWrapper'
import React from 'react'
import Component, { TextButton } from '.'

test(`${Component.name}-snapshot`, () => {
  const wrapper = mountWithTheme(<Component />)
  expect(wrapper).toMatchSnapshot()
})

test(`${Component.name}-event`, () => {
  const component = mountWithTheme(<Component />)
  component.find(TextButton).map(button => button.simulate('click'))
})
