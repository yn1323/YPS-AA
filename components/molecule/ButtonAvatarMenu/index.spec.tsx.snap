// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonAvatarMenu-snapshot 1`] = `
<ButtonAvatarMenu>
  <ButtonAvatarMenu__TextButton
    endIcon={<Memo />}
    onClick={[Function]}
  >
    <WithStyles(ForwardRef(Button))
      className="ButtonAvatarMenu__TextButton-sc-c2fxzv-0 gDCCjg"
      endIcon={<Memo />}
      onClick={[Function]}
    >
      <ForwardRef(Button)
        className="ButtonAvatarMenu__TextButton-sc-c2fxzv-0 gDCCjg"
        classes={
          Object {
            "colorInherit": "MuiButton-colorInherit",
            "contained": "MuiButton-contained",
            "containedPrimary": "MuiButton-containedPrimary",
            "containedSecondary": "MuiButton-containedSecondary",
            "containedSizeLarge": "MuiButton-containedSizeLarge",
            "containedSizeSmall": "MuiButton-containedSizeSmall",
            "disableElevation": "MuiButton-disableElevation",
            "disabled": "Mui-disabled",
            "endIcon": "MuiButton-endIcon",
            "focusVisible": "Mui-focusVisible",
            "fullWidth": "MuiButton-fullWidth",
            "iconSizeLarge": "MuiButton-iconSizeLarge",
            "iconSizeMedium": "MuiButton-iconSizeMedium",
            "iconSizeSmall": "MuiButton-iconSizeSmall",
            "label": "MuiButton-label",
            "outlined": "MuiButton-outlined",
            "outlinedPrimary": "MuiButton-outlinedPrimary",
            "outlinedSecondary": "MuiButton-outlinedSecondary",
            "outlinedSizeLarge": "MuiButton-outlinedSizeLarge",
            "outlinedSizeSmall": "MuiButton-outlinedSizeSmall",
            "root": "MuiButton-root",
            "sizeLarge": "MuiButton-sizeLarge",
            "sizeSmall": "MuiButton-sizeSmall",
            "startIcon": "MuiButton-startIcon",
            "text": "MuiButton-text",
            "textPrimary": "MuiButton-textPrimary",
            "textSecondary": "MuiButton-textSecondary",
            "textSizeLarge": "MuiButton-textSizeLarge",
            "textSizeSmall": "MuiButton-textSizeSmall",
          }
        }
        endIcon={<Memo />}
        onClick={[Function]}
      >
        <WithStyles(ForwardRef(ButtonBase))
          className="MuiButton-root MuiButton-text ButtonAvatarMenu__TextButton-sc-c2fxzv-0 gDCCjg"
          component="button"
          disabled={false}
          focusRipple={true}
          focusVisibleClassName="Mui-focusVisible"
          onClick={[Function]}
          type="button"
        >
          <ForwardRef(ButtonBase)
            className="MuiButton-root MuiButton-text ButtonAvatarMenu__TextButton-sc-c2fxzv-0 gDCCjg"
            classes={
              Object {
                "disabled": "Mui-disabled",
                "focusVisible": "Mui-focusVisible",
                "root": "MuiButtonBase-root",
              }
            }
            component="button"
            disabled={false}
            focusRipple={true}
            focusVisibleClassName="Mui-focusVisible"
            onClick={[Function]}
            type="button"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-text ButtonAvatarMenu__TextButton-sc-c2fxzv-0 gDCCjg"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                <ButtonAvatarMenu__StyledAvatar
                  alt="Remy Sharp"
                  src="/static/images/avatar/1.jpg"
                >
                  <WithStyles(ForwardRef(Avatar))
                    alt="Remy Sharp"
                    className="ButtonAvatarMenu__StyledAvatar-sc-c2fxzv-1 mVzJx"
                    src="/static/images/avatar/1.jpg"
                  >
                    <ForwardRef(Avatar)
                      alt="Remy Sharp"
                      className="ButtonAvatarMenu__StyledAvatar-sc-c2fxzv-1 mVzJx"
                      classes={
                        Object {
                          "circle": "MuiAvatar-circle",
                          "circular": "MuiAvatar-circular",
                          "colorDefault": "MuiAvatar-colorDefault",
                          "fallback": "MuiAvatar-fallback",
                          "img": "MuiAvatar-img",
                          "root": "MuiAvatar-root",
                          "rounded": "MuiAvatar-rounded",
                          "square": "MuiAvatar-square",
                        }
                      }
                      src="/static/images/avatar/1.jpg"
                    >
                      <div
                        className="MuiAvatar-root MuiAvatar-circular ButtonAvatarMenu__StyledAvatar-sc-c2fxzv-1 mVzJx"
                      >
                        <img
                          alt="Remy Sharp"
                          className="MuiAvatar-img"
                          src="/static/images/avatar/1.jpg"
                        />
                      </div>
                    </ForwardRef(Avatar)>
                  </WithStyles(ForwardRef(Avatar))>
                </ButtonAvatarMenu__StyledAvatar>
                <span
                  className="MuiButton-endIcon MuiButton-iconSizeMedium"
                >
                  <ForwardRef(KeyboardArrowDownIcon)>
                    <WithStyles(ForwardRef(SvgIcon))>
                      <ForwardRef(SvgIcon)
                        classes={
                          Object {
                            "colorAction": "MuiSvgIcon-colorAction",
                            "colorDisabled": "MuiSvgIcon-colorDisabled",
                            "colorError": "MuiSvgIcon-colorError",
                            "colorPrimary": "MuiSvgIcon-colorPrimary",
                            "colorSecondary": "MuiSvgIcon-colorSecondary",
                            "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                            "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                            "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                            "root": "MuiSvgIcon-root",
                          }
                        }
                      >
                        <svg
                          aria-hidden={true}
                          className="MuiSvgIcon-root"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
                          />
                        </svg>
                      </ForwardRef(SvgIcon)>
                    </WithStyles(ForwardRef(SvgIcon))>
                  </ForwardRef(KeyboardArrowDownIcon)>
                </span>
              </span>
              <WithStyles(memo)
                center={false}
              >
                <ForwardRef(TouchRipple)
                  center={false}
                  classes={
                    Object {
                      "child": "MuiTouchRipple-child",
                      "childLeaving": "MuiTouchRipple-childLeaving",
                      "childPulsate": "MuiTouchRipple-childPulsate",
                      "ripple": "MuiTouchRipple-ripple",
                      "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                      "rippleVisible": "MuiTouchRipple-rippleVisible",
                      "root": "MuiTouchRipple-root",
                    }
                  }
                >
                  <span
                    className="MuiTouchRipple-root"
                  >
                    <TransitionGroup
                      childFactory={[Function]}
                      component={null}
                      exit={true}
                    />
                  </span>
                </ForwardRef(TouchRipple)>
              </WithStyles(memo)>
            </button>
          </ForwardRef(ButtonBase)>
        </WithStyles(ForwardRef(ButtonBase))>
      </ForwardRef(Button)>
    </WithStyles(ForwardRef(Button))>
  </ButtonAvatarMenu__TextButton>
  <ListMenu
    anchorEl={null}
    delimeterPosition={
      Array [
        1,
      ]
    }
    items={
      Array [
        Object {
          "icon": <Memo />,
          "label": "設定",
          "link": "",
        },
        Object {
          "icon": <Memo />,
          "label": "よくある質問",
          "link": "",
        },
        Object {
          "icon": <Memo />,
          "label": "ログアウト",
          "link": "",
        },
      ]
    }
    placement="bottom-end"
    setShow={[Function]}
    show={false}
  >
    <ListMenu__Container
      anchorEl={null}
      open={false}
      placement="bottom-end"
      transition={true}
    >
      <ForwardRef(Popper)
        anchorEl={null}
        className="ListMenu__Container-sc-ib7wze-0 duhcwm"
        open={false}
        placement="bottom-end"
        transition={true}
      />
    </ListMenu__Container>
  </ListMenu>
</ButtonAvatarMenu>
`;
